src/app/utils/state.ts

```diff
@@ -28,21 +27,29 @@ class StateManager {
 
   public async init() {}
 
-  public async update() {
-    console.log("supabase url", process.env.SUPABASE_URL);
-    const status = await this.getLatestStatus();
-    console.log("tamagotchiStatus", status);
+  public async update(vectorSearchResult?: any[]) {
```

Comment:
**issue (complexity):** Consider splitting the update method into smaller focused methods to handle preferences, prompts and status updates separately

The update method can be simplified by extracting the prompt template and status processing logic. Consider refactoring like this:

```typescript
private getStatusPrompt(preferences: string, lastInteractions: string, status: any) {
  return PromptTemplate.fromTemplate(`
    ONLY return JSON as output. no prose.
    You are a virtual pet, about your preferences: 
    ${preferences}

    Here's the most recent interactions you had and their timestamps:
    {lastInteractions}

    Your previous status:
    {status}
    // ... rest of template ...
  `);
}

private async processPreferences(vectorSearchResult?: any[]) {
  return vectorSearchResult
    ? vectorSearchResult
        .map(item => item.content || "")
        .join("\n")
    : "No preferences";
}

public async update(vectorSearchResult?: any[]) {
  const statusData = await this.getLatestStatus();
  const preferences = await this.processPreferences(vectorSearchResult);
  const lastInteractions = await this.getInteractionsSince(statusData.updatedat);

  const prompt = this.getStatusPrompt(
    preferences,
    this.formatInteractions(lastInteractions),
    statusData.status
  );

  const chain = new LLMChain({ prompt, llm: this.model });
  const result = await chain.call({
    lastInteractions: this.formatInteractions(lastInteractions),
    status: statusData.status
  });

  await this.updateTamagotchiStatus({
    ...result.text,
    age: (statusData.status.age || 0) + 1
  });
}
```

This refactoring:
1. Extracts prompt template to a separate method
2. Moves preference processing to its own method
3. Simplifies the main update flow
4. Maintains all functionality while reducing nesting and improving readability